tests/test_parser.py

```diff
@@ -0,0 +1,1123 @@
+        return x
+
+    with pytest.raises(MyiaSyntaxError):
+        parse(f)
+
+
+def test_seq():
+    def f(x):  # pragma: nocover
+        x = x + 1
+        return 0
+
+    assert str_graph(parse(f)) == """graph f(%x) {
+  %_apply0 = <built-in function add>(%x, 1)
+  return 0
+}
+"""
+
+
+def test_seq2():
+    def f(x):  # pragma: nocover
+        return x + x
+
+    assert str_graph(parse(f)) == """graph f(%x) {
+  %_apply0 = <built-in function add>(%x, %x)
+  return %_apply0
+}
+"""
+
+def test_def():
+    def f():  # pragma: nocover
+        def g(a):
+            return a
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a) {
+  return %a
+}
+"""
+
+def test_def2():
+    def f():  # pragma: nocover
+        def g(a, *b):
+            return a
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a, %b) {
+  return %a
+}
+"""
+
+def test_def3():
+    def f():  # pragma: nocover
+        def g(a, b=1):
+            return b
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a, %b) {
+  return %b
+}
+"""
+
+def test_def4():
+    def f():  # pragma: nocover
+        def g(a, **b):
+            return a
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a, %b) {
+  return %a
+}
+"""
+
+
+def test_def6():
+    def f():  # pragma: nocover
+        def g(a: int) -> int:
+            return a
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a) {
+  return %a
+}
+"""
+
+
+def test_def5():
+    def f():  # pragma: nocover
+        def g(a, *, b):
+            return b
+        return g
+
+    assert str_graph(parse(f)) == """graph f() {
+  return @g
+}
+
+graph g(%a, %b) {
+  return %b
+}
+"""
+
+
+def test_getattr():
+    def f(a):  # pragma: nocover
+        return a.b
+
+    assert str_graph(parse(f)) == """graph f(%a) {
+  %_apply0 = <built-in function getattr>(%a, b)
+  return %_apply0
+}
+"""
+
+
+def test_binop():
+    def f(a, b):  # pragma: nocover
+        return a / b
+
+    assert str_graph(parse(f)) == """graph f(%a, %b) {
+  %_apply0 = <built-in function truediv>(%a, %b)
+  return %_apply0
+}
+"""
+
+
+def test_binop2():
+    def f(x, y):  # pragma: nocover
+        return x + y
+
+    assert str_graph(parse(f)) == """graph f(%x, %y) {
+  %_apply0 = <built-in function add>(%x, %y)
+  return %_apply0
+}
+"""
+
+
+def test_binop3():
+    def f(x, y):  # pragma: nocover
+        return x not in y
+
+    assert str_graph(parse(f)) == """graph f(%x, %y) {
+  %_apply0 = <built-in function contains>(%x, %y)
+  %_apply1 = <built-in function not_>(%_apply0)
+  return %_apply1
+}
+"""
+
+
+def test_boolop():
+    def f(a, b, c):  # pragma: nocover
+        return a and b or c
+
+    assert str_graph(parse(f)) == """graph f(%a, %b, %c) {
+  %_apply0 = universe_setitem(%_apply1, %b)
```

Comment:
What's `%_apply1`? Seems to be `make_handle` but it should appear here.

Reply:
This was fixed